### PR TITLE
Added View MQ (CRM + UI + tests)

### PR DESCRIPTION
**Pull request by hortha**
### Context

As a Support team user
I want to be able to view an MQ (Mandatory Qualification) award in TRS
So that I can check that the teaching record has the correct award allocated.

### Changes proposed in this pull request

Changes to UI and supporting backend to be able to View a person's Mandatory Qualifications and initate adding a new one.

### Guidance to review

CRM changes + UI + Tests.

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [ ] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by hortha** [comment]
[View an MQ in TRS](https://trello.com/c/FpOB2ghM/1247-view-an-mq-in-trs)

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/ReferenceDataCache.cs [hunk 1]
```diff
@@ -67,6 +67,12 @@ public async Task<dfeta_specialism> GetSpecialismByValue(string value)
         return specialisms.First(s => s.dfeta_Value == value);
     }
 
+    public async Task<dfeta_specialism> GetSpecialismById(Guid specialismId)
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/ReferenceDataCache.cs:70]:
```suggestion
    public async Task<dfeta_specialism> GetMqSpecialismById(Guid specialismId)
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Qualifications.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        var qualifications = await _crmQueryDispatcher.ExecuteQuery(new GetQualificationsByContactIdQuery(PersonId));
+        MandatoryQualifications = await MapMandatoryQualifications(qualifications!);
+
+        return Page();
+    }
+
+    private async Task<MandatoryQualificationInfo[]> MapMandatoryQualifications(dfeta_qualification[] qualifications)
+    {
+        var mandatoryQualifications = new List<MandatoryQualificationInfo>();
+        foreach (var qualification in qualifications.Where(q => q.dfeta_Type == dfeta_qualification_dfeta_Type.MandatoryQualification).OrderByDescending(q => q.dfeta_CompletionorAwardDate))
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Qualifications.cshtml.cs:61]: Could use LINQ here and avoid the `List<>` and `foreach`

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/ReferenceDataCache.cs (hunk 1 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/ReferenceDataCache.cs]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Qualifications.cshtml.cs (hunk 2 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Qualifications.cshtml.cs]: Done

**Review by gunndabad — APPROVED** [review]

**Pull request merged**